src/arrayMethodSort.js

```diff
@@ -4,8 +4,32 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (this.length <= 1) {
+      return this;
+    }
+
+    const defaultCompare = (a, b) => {
+      const aStr = String(a);
+      const bStr = String(b);
+
+      return aStr > bStr ? 1 : aStr < bStr ? -1 : 0;
+    };
+
+    const compare = compareFunction || defaultCompare;
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - i - 1; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          const temp = this[j];
+
+          this[j] = this[j + 1];
+          this[j + 1] = temp;
+        }
+      }
+    }
```

Comment:
The bubble sort algorithm is correctly implemented here. However, note that bubble sort is not the most efficient sorting algorithm for large datasets. Consider using more efficient algorithms like quicksort or mergesort if performance becomes an issue.